cmd/globalAlignment/globalAlignment.go

```diff
@@ -8,17 +8,40 @@ import (
 	"github.com/vertgenlab/gonomics/fileio"
 	"github.com/vertgenlab/gonomics/simpleGraph"
 	"log"
+	"strings" //raven added this line for CountSeqIdx
+	"os" //raven added this line for MSA Fasta output
 )
 
+//raven edited this block to specify only 1 sequnce is expected in each fasta file
 func usage() {
-	fmt.Print("./globalAlignment - chelsea's global alignment \n")
+	fmt.Print(
+	"./globalAlignment - chelsea's global alignment\n" +
+	"Usage:\n" +
+	" Align 2 .fasta files, each with only 1 sequence\n" +
+	"options:\n")
 	flag.PrintDefaults()
 }
 
+//raven wrote this block to count sequences based on the Read function in gonomics/fasta/fasta.go
+//maybe there are existing functions that do this already? I didn't find an existing function that returns seqIdx
+func CountSeqIdx(filename string) int {
+	var line string
+	var seqIdx int = 0 //I know in Read seqIdx is int64 and starts with -1, but I am using it differently here
+	var doneReading bool = false
+	file := fileio.EasyOpen(filename)
+	defer file.Close()
+	for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+		if strings.HasPrefix(line, ">") {
+			seqIdx++
+		}
+	}
+	return seqIdx
+}
+
 func main() {
-	var expectedNum int = 2
-	flag.Usage = usage
+	faOut := flag.String("faOut","globalAlignment_result.fa","name of the MSA output file") //raven added this line
```

Comment:
Some people may want the graph output instead of the .fa output.  I would consider making the default value the blank string ("") and then adding a check where you only write the file if the filename is not "".

Reply:
I like this suggestion and changed my code accordingly